### PR TITLE
Last step for siren authorisation

### PR DESCRIPTION
**Pull request by NorfairKing**

### CODE REVIEW

File: apps/hs/helen/src/Helen/Core/Types.hs [hunk 1]
```diff
@@ -58,16 +61,51 @@ import           Control.Monad.Trans.Control
...
+        Set.union
+          (confTrustedClients c1)
+          (confTrustedClients c2)
+    }
+
+instance FromJSON Config where
+  parseJSON = withObject "Config" $ \o -> do
+    tclos <- o .: "trusted-clients"
+    tcls <- forM tclos $ \tclo -> do
+      portstr <- (tclo .: "port")
```
Comment (achirkin) [inline apps/hs/helen/src/Helen/Core/Types.hs]: Don't clients connect using the same port all the time?
So far, helen specifies the port she listens to, and all clients just connect to this port from various ip addresses.
Therefore, I think there is no point to keep both, port and host, for each client in JSON/yaml configuration.
Maybe, let's just keep them as a list of strings? (each string is an ip address).

File: apps/hs/helen/src/Helen/Core.hs [hunk 2]
```diff
@@ -79,9 +89,38 @@ initHelen = do
...
+    , trustedClients = confTrustedClients conf
     }
 
+readConfigFromYamlFile :: IO Config
+readConfigFromYamlFile = do
+  cf <- resolveFile' "helen-config.yaml"
+  errOrConfig <- readYamlSafe cf
+  case errOrConfig of
+    Left err -> do
+      putStrLn $ unlines
```
Comment (achirkin) [inline apps/hs/helen/src/Helen/Core.hs:98]: Do you mean `HelenWorld` cannot be initialized before you run `readConfigFromYamlFile`, and that is the reason you don't use MonadLogger here?

File: apps/hs/helen/src/Helen/Core.hs [hunk 3]
```diff
@@ -79,9 +89,38 @@ initHelen = do
...
+    Nothing -> pure Set.empty
+    Just clientStr -> fmap Set.fromList $ forM (words clientStr) $ \cs ->
+      case splitOn ":" cs of
+        [ipstr, portstr] -> do
+          ipAddr <- Network.inet_addr ipstr
+          port <- case readMaybe portstr of
+            Nothing -> die $ "Failed to parse port from " <> portstr
+            Just p -> pure p
+          pure $ Network.SockAddrInet port ipAddr
+        _ -> die $ "Failed to parse ip address from " <> cs
```
Comment (achirkin) [inline apps/hs/helen/src/Helen/Core.hs]: Please, let's avoid halting program if there is any single chance to continue somehow with the error message sent to log :) Failing to parse ip address is not so big deal :)

**Review by achirkin — CHANGES_REQUESTED** [review]
`luci-connect` is fine (apart from that I would ask you to document top-level exported functions).

In `helen`, I added several comments regarding config files.
Have not checked `qua-server` part yet.

File: apps/hs/helen/src/Helen/Core.hs (hunk 2 above)
Comment (NorfairKing) [reply apps/hs/helen/src/Helen/Core.hs:98]: That's right.

File: apps/hs/helen/src/Helen/Core.hs (hunk 3 above)
Comment (NorfairKing) [reply apps/hs/helen/src/Helen/Core.hs]: Okido!

File: apps/hs/helen/src/Helen/Core/Types.hs (hunk 1 above)
Comment (NorfairKing) [reply apps/hs/helen/src/Helen/Core/Types.hs]: Also fixed. We'll be keeping an 'IP' instead of a string.

**Pull request merged**